plugins/com.google.cloud.tools.eclipse.appengine.facets/src/com/google/cloud/tools/eclipse/appengine/facets/FacetInstallDelegate.java

```diff
@@ -1,114 +1,124 @@
+    for (int i = 0, length = rawClasspath.length; i < length; i++) {
+      if (rawClasspath[i].equals(appEngineContainer)) {
+        return;
       }
-
-      project.addTargetedRuntime(appEngineFacetRuntime, monitor);
-      project.setPrimaryRuntime(appEngineFacetRuntime, monitor);
     }
-  }
-
-  public static void installAppEngineFacet(IFacetedProject facetedProject, IProgressMonitor monitor)
-      throws CoreException {
-    IFacetedProjectWorkingCopy workingCopy = facetedProject.createWorkingCopy();
-    IProjectFacet appEngineFacet = ProjectFacetsManager.getProjectFacet(AppEngineStandardFacet.ID);
-    IProjectFacetVersion appEngineFacetVersion = appEngineFacet.getVersion(AppEngineStandardFacet.VERSION);
-    workingCopy.addProjectFacet(appEngineFacetVersion);
-    workingCopy.commitChanges(monitor);
-  }
 
-  private void updateClasspath(IProject project, IProgressMonitor monitor) throws CoreException {
-    IJavaProject javaProject = JavaCore.create(project);
-    IClasspathEntry[] rawClasspath = javaProject.getRawClasspath();
     IClasspathEntry[] newClasspath = new IClasspathEntry[rawClasspath.length + 1];
     System.arraycopy(rawClasspath, 0, newClasspath, 0, rawClasspath.length);
-    newClasspath[newClasspath.length - 1] =
-        JavaCore.newContainerEntry(new Path(AppEngineSdkClasspathContainer.CONTAINER_ID),
-                                   new IAccessRule[0],
-                                   new IClasspathAttribute[]{
-                                       UpdateClasspathAttributeUtil.createDependencyAttribute(true /*isWebApp */)
-                                   },
-                                   true /* isExported */);
+    newClasspath[newClasspath.length - 1] = appEngineContainer;
     javaProject.setRawClasspath(newClasspath, monitor);
   }
 
+  /**
+   * Creates an appengine-web.xml file in the WEB-INF folder if it doesn't exist
+   */
+  private static void createConfigFiles(IProject project, IProgressMonitor monitor)
+      throws CoreException {
+    IFile appEngineWebXml = project.getFile(APPENGINE_WEB_XML_PATH);
+    if (appEngineWebXml.exists()) {
+      return;
+    }
+
+    IFolder configDir = project.getFolder(APPENGINE_WEB_XML_DIR);
+    if (!configDir.exists()) {
+      Path configDirPath = new Path(APPENGINE_WEB_XML_DIR);
+      IContainer current = project;
+      for (int i = 0; i < configDirPath.segmentCount(); i++) {
+        final String segment = configDirPath.segment( i );
+        IFolder folder = current.getFolder(new Path(segment));
+
+        if (!folder.exists()) {
+          folder.create( true, true, monitor );
+        }
+        current = folder;
+      }
+      configDir = (IFolder) current;
+    }
+
+    InputStream in = FacetInstallDelegate.class.getResourceAsStream("templates/" + APPENGINE_WEB_XML + ".ftl");
+    if (in == null) {
+      IStatus status = StatusUtil.error(FacetInstallDelegate.class,
+          "Could not load template for " + APPENGINE_WEB_XML);
```

Comment:
There are a bunch of messages that probably should be moved to the message bundle. Will do this in another PR. Created an issue for this #494